process/smartContract/hooks/blockChainHook.go

```diff
@@ -433,13 +428,19 @@ func (bh *BlockChainHookImpl) getUserAccounts(
 		}
 	}
 
-	acc, err := bh.accounts.LoadAccount(dstAddr)
-	if err != nil {
-		return nil, nil, err
-	}
-	dstAccount, ok := acc.(state.UserAccountHandler)
-	if !ok {
-		return nil, nil, process.ErrWrongTypeAssertion
+	var dstAccount state.UserAccountHandler
+	dstShardId := bh.shardCoordinator.ComputeId(dstAddr)
+	if dstShardId == bh.shardCoordinator.SelfId() {
+		acc, err := bh.accounts.LoadAccount(dstAddr)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		var ok bool
+		dstAccount, ok = acc.(state.UserAccountHandler)
+		if !ok {
+			return nil, nil, process.ErrWrongTypeAssertion
+		}
 	}
 
 	return sndAccount, dstAccount, nil
```

Comment:
you return nil for the dstAccount if the dstAccount is not in the current shard. Is this the desired behavior?

Reply:
yes.